### PR TITLE
Fix defence ani (fix mantis 0001936)

### PR DESCRIPTION
**Pull request by edeksumo**

### CODE REVIEW

**Comment by ArseniyShestakov** [comment]
Welcome! Nice to see you managed to win in fight over build issues. :-)
Here is few advices unrelated to code:

* Most of bugs already reported on bug tracker so if you fix something try to find and comment on issues you're about to fix. E.g I suppose this one is relevant for your pull request:
http://bugs.vcmi.eu/view.php?id=1936
* Check code formatting guide and try to stick it, you can as well look on @alexvins spells code as example:
http://wiki.vcmi.eu/index.php?title=Coding_guidelines
There is a lot of old code that with bad formatting (spaces instead of tabs, spaces before brackets in conditions, etc) so you can also fix formatting around code you work on too. Or at least fix it in lines you change.

Thanks for reading!

**Comment by edeksumo** [comment]
I'm too happy, now I can help you guys with VCMI. I will update formating. But first I have to know more about github, and editing pull request :)

**Comment by ArseniyShestakov** [comment]
BTW i'm not aware how that code works, but is there any particular reason why you need to add own EBattleStackState state instead of just checking "DEFENING"?

Also feels like selector was intend to actually do something even if it's didn't work. E.g it's last time edited by Warmonger at b925a75eccc2c565e176478a8d36f8ad2c27c3b3.

**Comment by ArseniyShestakov** [comment]
Probably Warmonger's code should be checking for "DEFENSIVE_STANCE" bonus instead of primary skill. Still if this selector needed in animation code at all depend on fact if there any creatures that should show defense animation without getting defend command in battle.

**Comment by edeksumo** [comment]
It wasn't work :), when I was using "DEFENING" state creature defend animation work only to end of round, and after that there was only hit ani. So it was the simplest way to fix it.

**Comment by ArseniyShestakov** [comment]
Then fact that "DEFENDING" stance not remain after round it's what need to be fixed. ;-)

**Comment by edeksumo** [comment]
I try to change it, but when i move it just before taking action by creature, thats made that creature never get turn till battle end

**Comment by DjWarmonger** [comment]
So few remarks:
- The common abbreviation for "animation" in teh project is "anim", not "ani".
- In the future describe the issue in "description" field, easier to figure out what's going on
- Change in animations shouldn't cause change in gameplay mechanics

>> Probably Warmonger's code should be checking for "DEFENSIVE_STANCE" bonus instead of primary skill

I don't have anythingto do with animations.

**Comment by edeksumo** [comment]
So I change ANI to ANIM, now can it be merged? Or I have to change something in this? :)

File: lib/NetPacksLib.cpp [hunk 1]
```diff
@@ -1304,6 +1304,9 @@ DLL_LINKAGE void StartAction::applyGs( CGameState *gs )
 {
 	CStack *st = gs->curB->getStack(ba.stackNumber);
 
+	if(vstd::contains(st->state, EBattleStackState::DEFENDING_ANIM))
```
Comment (ArseniyShestakov) [inline lib/NetPacksLib.cpp]: This one cause crashes:

* When quick battle is enabled (no animations going to happen at all).
* When spell casted on first turn in battle.

**Comment by edeksumo** [comment]
Ok, I will fix it soon.

**Pull request merged**